python/cugraph/centrality/katz_centrality.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from cugraph.centrality import katz_centrality_wrapper
+
+
+def katz_centrality(G,
+                    alpha=0.1,
+                    max_iter=100,
+                    tol=1.0e-5,
+                    nstart=None,
+                    normalized=True):
+    """
+    Compute the Katz centrality for the nodes of the graph G
+
+    Parameters
+    ----------
+    graph : cuGraph.Graph
+        cuGraph graph descriptor with connectivity information. Edge weights,
+        if present, should be single or double precision floating point values.
+    alpha : float
+        Attenuation factor with a default value of 0.1. Alpha is set to
+        1/(lambda_max) if it is greater where lambda_max is the maximum degree
```

Comment:
Just to double check, it is "greater" or it is smaller (i.e. shouldn't alpha <= 1 / lambda_max to converge? so, if alpha is smaller than 1/lambda_max, no need to change alpha but if alpha > 1 / lambda_max, shouldn't we reset alpha to 1 / lambda_max?). And if alpha is too large, shouldn't we better throw exceptions rather than silently (even though documented) resetting alpha to another value?

Reply:
> Alpha is set to 1/(lambda_max) if it is greater

I am implying that Alpha = min(alpha, 1/lambda_max)

> And if alpha is too large, shouldn't we better throw exceptions rather than silently (even though documented) resetting alpha to another value?

That could be done. We are computing out_degree's of the graph to reset this value. If we give up this responsibility to the user that would mean lesser computation for us.

Reply:
I'm not saying we should give up this responsibility, but once we detected that alpha is too large, we'd better throw exception instead of silently setting alpha to a new value as this may surprise the user.

Reply:
We still need to check that alpha is small enough for convergence, so I don't think this will reduce the amount of computing.

Reply:
Fixed. `gdf_katz_centrality` will return `GDF_CUDA_ERROR` along with a message notifying the user of the largest acceptable `alpha` value for invalid user input.